src/main.cpp

```diff
@@ -3036,7 +3036,11 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
             sapling_tree.append(outputDescription.cmu);
         }
 
-        orchard_tree.AppendBundle(tx.GetOrchardBundle());
+        if (!orchard_tree.AppendBundle(tx.GetOrchardBundle())) {
+            return state.DoS(100,
+                error("ConnectBlock(): the Orchard commitment tree is full."),
+                REJECT_INVALID, "orchard-commitment-tree-full");
+        };
```

Comment:
For Sprout and Sapling, we simply throw an error if the tree is full instead of rejecting the block. It has the same effect, but I think a rejection here is clearer (and allows the chain to progress without Orchard components).